quinn-proto/src/connection/mod.rs

```diff
@@ -796,6 +797,44 @@ impl Connection {
                 break;
             }
 
+            // Send an off-path PATH_RESPONSE. LIFO since old challenges might have been forgotten by the sender.
+            if space_id == SpaceId::Data
+                && num_datagrams == 1
+                && self
+                    .path_responses
+                    .last()
+                    .map_or(false, |(_, remote)| *remote != self.path.remote)
+            {
+                let (response, remote) = self.path_responses.pop().unwrap();
+                let mut builder = builder_storage.take().unwrap();
```

Comment:
How do we know these `unwrap`s to be safe? Should be use a nested `if let Some(..) = self.path_responses.last() { if space_id == SpaceId::Data && .. {` to avoid the first one?

Reply:
Removed one by refactoring, and documented the other.